## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 520316`
Commit: [`66ee29a63fb20f9deef904856ba9ff0ee37af3fb`](https://github.com/NixOS/nixpkgs/commit/66ee29a63fb20f9deef904856ba9ff0ee37af3fb) ([subsequent changes](https://github.com/NixOS/nixpkgs/compare/66ee29a63fb20f9deef904856ba9ff0ee37af3fb..pull/520316/head))
Merge: [`ee8ad9e47d5115003382888e9389cfc141bbc969`](https://github.com/NixOS/nixpkgs/commit/ee8ad9e47d5115003382888e9389cfc141bbc969)

Logs: https://github.com/Renna42/nixpkgs-review-gha/actions/runs/25903103717


---
### `x86_64-linux`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>openscreen</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-linux`</summary>
<details>
<summary>openscreen</summary>
<pre>  • executing @electron/rebuild  electronVersion=41.3.0 arch=x64 buildFromSource=true workspaceRoot=/build/source projectDir=./ appDir=./
  • installing native dependencies  arch=x64
  • preparing       moduleName=uiohook-napi arch=x64
  ⨯ make: Entering directory &#x27;/build/source/node_modules/uiohook-napi/build&#x27;
  CC(target) Release/obj.target/libuiohook/libuiohook/src/logger.o
  CC(target) Release/obj.target/libuiohook/libuiohook/src/x11/input_helper.o
../libuiohook/src/x11/input_helper.c:23:10: fatal error: X11/keysym.h: No such file or directory
   23 | #include &lt;X11/keysym.h&gt;
      |          ^~~~~~~~~~~~~~
compilation terminated.
make: *** [libuiohook.target.mk:126: Release/obj.target/libuiohook/libuiohook/src/x11/input_helper.o] Error 1
make: Leaving directory &#x27;/build/source/node_modules/uiohook-napi/build&#x27;
Error: `make` failed with exit code: 2
    at ChildProcess.&lt;anonymous&gt; (/build/source/node_modules/node-gyp/lib/build.js:219:23)


  ⨯ node-gyp failed to rebuild &#x27;/build/source/node_modules/uiohook-napi&#x27;  failedTask=build stackTrace=Error: node-gyp failed to rebuild &#x27;/build/source/node_modules/uiohook-napi&#x27;
    at ChildProcess.&lt;anonymous&gt; (file:///build/source/node_modules/@electron/rebuild/lib/module-type/node-gyp/node-gyp.js:114:24)
    at ChildProcess.emit (node:events:508:28)
    at ChildProcess._handle.onexit (node:internal/child_process:294:12)</pre>
</details>
</details>

---
### `aarch64-linux`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>openscreen</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-linux`</summary>
<details>
<summary>openscreen</summary>
<pre>  • executing @electron/rebuild  electronVersion=41.3.0 arch=arm64 buildFromSource=true workspaceRoot=/build/source projectDir=./ appDir=./
  • installing native dependencies  arch=arm64
  • preparing       moduleName=uiohook-napi arch=arm64
  ⨯ make: Entering directory &#x27;/build/source/node_modules/uiohook-napi/build&#x27;
  CC(target) Release/obj.target/libuiohook/libuiohook/src/logger.o
  CC(target) Release/obj.target/libuiohook/libuiohook/src/x11/input_helper.o
../libuiohook/src/x11/input_helper.c:23:10: fatal error: X11/keysym.h: No such file or directory
   23 | #include &lt;X11/keysym.h&gt;
      |          ^~~~~~~~~~~~~~
compilation terminated.
make: *** [libuiohook.target.mk:124: Release/obj.target/libuiohook/libuiohook/src/x11/input_helper.o] Error 1
make: Leaving directory &#x27;/build/source/node_modules/uiohook-napi/build&#x27;
Error: `make` failed with exit code: 2
    at ChildProcess.&lt;anonymous&gt; (/build/source/node_modules/node-gyp/lib/build.js:219:23)


  ⨯ node-gyp failed to rebuild &#x27;/build/source/node_modules/uiohook-napi&#x27;  failedTask=build stackTrace=Error: node-gyp failed to rebuild &#x27;/build/source/node_modules/uiohook-napi&#x27;
    at ChildProcess.&lt;anonymous&gt; (file:///build/source/node_modules/@electron/rebuild/lib/module-type/node-gyp/node-gyp.js:114:24)
    at ChildProcess.emit (node:events:508:28)
    at ChildProcess._handle.onexit (node:internal/child_process:294:12)</pre>
</details>
</details>

---
### `x86_64-darwin` (sandbox = relaxed)
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>openscreen</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-darwin`</summary>
<details>
<summary>openscreen</summary>
<pre>  • loaded configuration  file=/nix/build/nix-58290-2345705359/source/electron-builder.json5
  • description is missed in the package.json  appPackageFile=/nix/build/nix-58290-2345705359/source/package.json
  • @electron/rebuild already used by electron-builder, please consider to remove excess dependency from devDependencies

To ensure your native dependencies are always matched electron version, simply add script `&quot;postinstall&quot;: &quot;electron-builder install-app-deps&quot; to your `package.json`
  • writing effective config  file=release/1.4.0/builder-effective-config.yaml
  • executing @electron/rebuild  electronVersion=41.3.0 arch=x64 buildFromSource=true workspaceRoot=/nix/build/nix-58290-2345705359/source projectDir=./ appDir=./
  • installing native dependencies  arch=x64
  • preparing       moduleName=uiohook-napi arch=x64
  ⨯ gyp: No Xcode or CLT version detected!
Error: `gyp` failed with exit code: 1
    at ChildProcess.&lt;anonymous&gt; (/nix/build/nix-58290-2345705359/source/node_modules/node-gyp/lib/configure.js:317:18)
    at ChildProcess.emit (node:events:508:28)
    at ChildProcess._handle.onexit (node:internal/child_process:294:12)


  ⨯ node-gyp failed to rebuild &#x27;/nix/build/nix-58290-2345705359/source/node_modules/uiohook-napi&#x27;  failedTask=build stackTrace=Error: node-gyp failed to rebuild &#x27;/nix/build/nix-58290-2345705359/source/node_modules/uiohook-napi&#x27;
    at ChildProcess.&lt;anonymous&gt; (file:///nix/build/nix-58290-2345705359/source/node_modules/@electron/rebuild/lib/module-type/node-gyp/node-gyp.js:114:24)
    at ChildProcess.emit (node:events:508:28)
    at ChildProcess._handle.onexit (node:internal/child_process:294:12)</pre>
</details>
</details>

---
### `aarch64-darwin` (sandbox = relaxed)
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>openscreen</li>
  </ul>
</details>

--- a/pkgs/by-name/op/openscreen/package.nix
+++ b/pkgs/by-name/op/openscreen/package.nix
@@ -6,33 +6,38 @@
   copyDesktopItems,
   makeWrapper,
   makeDesktopItem,
-  electron_39,
+  xcbuild,
+  electron_41,
   nodejs_22,
   nix-update-script,
-
-  nodejs ? nodejs_22,
-  electron ? electron_39,
 }:
+let
+  electron = electron_41;
+  nodejs = nodejs_22;
+in
 buildNpmPackage (finalAttrs: {
   inherit nodejs;
 
   pname = "openscreen";
-  version = "1.3.0";
+  version = "1.4.0";
 
   src = fetchFromGitHub {
     owner = "siddharthvaddem";
     repo = "openscreen";
     tag = "v${finalAttrs.version}";
-    hash = "sha256-0L+SGbhKNvSH5IpjHjS5G1uhnoWV5oIaSs2mRSAl/mM=";
+    hash = "sha256-ZBWDQVYDXJ/IQGhlmscmCOMjpl03kVIdMoJXOW8OjUI=";
   };
 
-  npmDepsHash = "sha256-pKfOxRzVfMNVHxA9oM9naWz024api8jxiTJwy0+6W9A=";
+  npmDepsHash = "sha256-SMAYgOwlZg9+/KZBUhVviOxEdMeL3Z2YdC8Hx8Q/ioY=";
 
   npmRebuildFlags = [ "--ignore-scripts" ]; # Prevent running `node-gyp build`
 
   nativeBuildInputs = [
     makeWrapper
     copyDesktopItems
+  ]
+  ++ lib.optionals (stdenv.hostPlatform.isDarwin && stdenv.hostPlatform.isx86_64) [
+    xcbuild
   ];
 
   env.ELECTRON_SKIP_BINARY_DOWNLOAD = "1";
@@ -60,7 +65,8 @@ buildNpmPackage (finalAttrs: {
       npm exec electron-builder -- \
         --dir \
         -c.electronDist=${electron.dist} \
-        -c.electronVersion=${electron.version}
+        -c.electronVersion=${electron.version} \
+        -c.npmRebuild=false
     ''}
 
     runHook postBuild

